app/src/webview/java/org/mozilla/focus/webview/TrackingProtectionWebViewClient.java

```diff
@@ -137,4 +140,28 @@ public void onPageStarted(WebView view, String url, Bitmap favicon) {
 
         super.onPageStarted(view, url, favicon);
     }
+
+    @Override
+    public void onReceivedHttpAuthRequest(WebView view, final HttpAuthHandler handler, String host, String realm) {
+
+        httpAuthenticationDialog = new HttpAuthenticationDialog(context, host, realm);
+        httpAuthenticationDialog.setOkListener(new HttpAuthenticationDialog.OkListener() {
+            @Override
+            public void onOk(String host, String realm, String username, String password) {
+                handler.proceed(username, password);
+            }
+        });
+        httpAuthenticationDialog.setCancelListener(new HttpAuthenticationDialog.CancelListener() {
+            @Override
+            public void onCancel() {
+                handler.cancel();
+                httpAuthenticationDialog = null;
+            }
+        });
+
+        httpAuthenticationDialog.show();
+    }
```

Comment:
This is the wrong place for this. IWebView.Callback should provide a generic on** method for that and this is the method we should call here. Background is: There are multiple implementations of IWebView and we do. not want to add http-auth-dialogs to all of them. So instead we pass this along. With that we only need to implement the http-auth-dialog in one place for all implementations of IWebView. Luckily the callback will be executed in a fragment and with that you do not need all the passing around of the Context here.